sa/sa.go

```diff
@@ -511,8 +529,16 @@ func (ssa *SQLStorageAuthority) MarkCertificateRevoked(ctx context.Context, seri
 
 // UpdateRegistration stores an updated Registration
 func (ssa *SQLStorageAuthority) UpdateRegistration(ctx context.Context, reg core.Registration) error {
-	var rm regModel
-	err := ssa.dbMap.SelectOne(&rm, fmt.Sprintf("SELECT %s FROM registrations WHERE id = %d", regFields, reg.ID))
+	var regType interface{}
+	var fields string
+	if features.Enabled(features.AllowAccountDeactivation) {
+		regType = &regModelv2{}
+		fields = regV2Fields
+	} else {
+		regType = &regModelv1{}
+		fields = regV1Fields
+	}
+	err := ssa.dbMap.SelectOne(regType, fmt.Sprintf("SELECT %s FROM registrations WHERE id = %d", fields, reg.ID))
```

Comment:
Can you use a `?` substitution for the `reg.ID` instead of making it part of the sprintf for the `fields`?